### DISCUSSION TITLE
Possible to monitor CPU and RAM of LXC in Proxmox?

### DISCUSSION THREAD

**Discussion by dimatx** [henrygd/beszel#65] [Support]
When I setup the beszel-agent in my LXC running on Proxmox, the CPU and RAM that are shown reflect the host, rather than the Linux container running on it. I figure that's what LXC is reporting back, but wondering if there is a way to get the CPU and RAM to reflect the cores/ram/usage for the container, rather than the host running the container. Disk space does report just what the container has allocated.

In the below screenshot, proxmox is running on the host, and docker is the name of an LXC container running on the host.

![image](https://github.com/user-attachments/assets/09018651-b5db-4c9b-a7a1-aef36e973581)

**Comment by utlilb** [comment]
I was just getting ready to add a couple LXC agents. I'll let you know if I have the same issue.

**Comment by henrygd** [comment]
Probably, I'll look into it when I have time.

**Reply by dimatx** [reply]
Thanks!

If it's helpful, Dozzle is an example that does show the actual cores/memory allocated to the LXC rather than what the host has.

![image](https://github.com/user-attachments/assets/33cbff35-364c-4fb9-b71b-0b240ea54176)

**Comment by tymanthius** [comment]
I know this is old, but I just set this up on Unraid, and I'm seenig the same issue.  LXC shows the same (w/in about 0.2%) for memory and cpu as the host.  But Unraid reports a more accurate 'very low' usage for the container as it's not doing much of anything right now.

**Reply by henrygd** [reply]
Please check #176 for more on this issue.

**Comment by cryptochrome** [comment]
This is how LXC fundamentally works. LXC containers don't have their "own" memory like a virtual machine would have. An LXC container is just a namespace on the host, similar to chroot. The container shares the same hardware (including memory) with the host. Even if you set a limit, the container will still see the full amount of RAM of the host (the limit just makes sure the container can't use more than what you have assigned). 

In other words: Trying to monitor RAM usage inside a container is more or less pointless. 

If you wanted to monitor how much of the host's total RAM it consumes, Beszel would need to "understand" LXC, which, AFAIK, it does not do, currently.
